tools/SourceKit/include/SourceKit/Core/Context.h

```diff
@@ -51,12 +54,41 @@ class GlobalConfig {
   Settings::CompletionOptions getCompletionOpts() const;
 };
 
+class SlowRequestSimulator {
+  std::map<SourceKitCancellationToken, std::shared_ptr<Semaphore>>
```

Comment:
This needs a lock since it's being modified on the concurrent request handling queue.

Reply:
Good catch. Thanks.